File: PillarsDivTwo/solve/java/PillarsDivTwoSolver.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class PillarsDivTwoSolver {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        List<Integer> heightBoxed = (List<Integer>) reader.next(new TypeRef<List<Integer>>(){}.getType());
        int[] height = new int[heightBoxed.size()];
        for (int _i = 0; _i < heightBoxed.size(); ++_i)
            height[_i] = heightBoxed.get(_i);
        reader.next();
        
        int w = (Integer) reader.next(Integer.class);
        reader.close();

        PillarsDivTwo solver = new PillarsDivTwo();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.maximalLength(height, w));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }
}

File: PillarsDivTwo/solve/cpp/PillarsDivTwoSolver.cpp
#include <fstream>
#include <topcoder>
#include "PillarsDivTwo.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
  try {
    std::ifstream ifs(argv[1]);
    vector<int> height; tc::read(ifs, height); tc::next(ifs);
    int w; tc::read(ifs, w);
    ifs.close();

    std::ofstream ofs(argv[2]);
    PillarsDivTwo solver;
    tc::show(ofs, solver.maximalLength(height, w));
    ofs.close();
  } catch (std::exception &e) {
    std::cerr << e.what() << std::endl;
  }
  return 0;
}
